metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_queries.py

```diff
@@ -654,7 +657,14 @@ def _build_enriched_query_log_query(
 fingerprinted_queries as (
     SELECT *,
         -- TODO: Generate better fingerprints for each query by pushing down regex logic.
-        query_history.query_parameterized_hash as query_fingerprint
+        query_history.query_parameterized_hash as query_fingerprint,
+        -- Optional and additional hash to be used for query deduplication and final query identity
+        CASE 
+            WHEN CONTAINS(query_history.query_text, '-- Hex query metadata:')
+            -- Extract project id and hash it
+            THEN CAST(HASH(REGEXP_SUBSTR(query_history.query_text, '"project_id"\\\\s*:\\\\s*"([^"]+)"', 1, 1, 'e', 1)) AS VARCHAR)
```

Comment:
we probably need to include other things beyond just the project id

Reply:
There should be no reason to add additional field to the hash because, as soon as we have the project_id, we are fine to build the lineage... we don't care if it's scheduled run or not.

However, in https://github.com/datahub-project/datahub/pull/13126 I'm filtering out non-scheduled runs. Which invalidates previous rationale, we need to differentiate scheduled runs from non-scheduled runs, so we can properly filter them out later in hex ingestor. Otherwise (if not considering context in the hash), the scheduled run could be deduplicated and lost here and leading to lineage being lost too.

Conclusion: I'm adding `context` to this hash.

Reply:
addressed in commit 3907f84e6f1ef1cce83fc3a6dc6c5c7ad8a7638b